dist/tools/compile_test/tests/test.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+COMPILE_TEST_DIR="$(readlink -f "$(dirname "$0")/..")"
+
+# prints a character representing the result:
+#   - a green tick when $1 is 0
+#   - a red cross when $1 is 1
+function print_result {
+    local RED="\033[0;31m"
+    local GREEN="\033[0;32m"
+    local NO_COLOUR="\033[0m"
+
+    if (( "$1" == 0 )); then
+        echo -e "${GREEN}✓$NO_COLOUR"
+    else
+        echo -e "${RED}x$NO_COLOUR"
+    fi
+}
+
+
+# sets the global accumulative RESULT variable to keep record of errors during
+# tests.
+function set_result {
```

Comment:
Try naming the function after its purpose, not what it actually does. e.g. "accumulate_errors".

Reply:
Done